src/pdf.rs

```diff
@@ -358,10 +228,14 @@ mod test {
     #[test]
     fn ref_writer() {
         let mut file = tempfile().expect("tempfile failed");
-        let surface = RefWriter::new(100., 100., &mut file);
+        let surface = unsafe {
+            PdfSurface::for_raw_stream(100., 100., &mut file)
+        };
 
         draw(&surface);
         surface.finish();
+        surface.take_output_stream();
+        drop(file);
     }
```

Comment:
Since we discussed with @sdroege that writing to a stream after `finish()` would result in an error, it would be nice to perhaps test for that as well, to see if it's handled gracefully and stuff.

Reply:
Err, actually it looks like this isn't implemented, looks like `write_callback()` does nothing if the stream is taken. This is what I originally suggested, but I think @sdroege is right in that this should instead be an error. Also, seeing the code in tests, it occurs to me that it might make sense to call `finish()` as part of `take_output_stream()`. Does it make sense to call `take_output_stream()` without calling `finish()` right before that? I can't think of a situation where that would make sense...

Reply:
Ok, sorry for the fragmented comments, but it seems like cairo [already handles](https://www.cairographics.org/manual/cairo-cairo-surface-t.html#cairo-surface-finish) this by setting an error state if drawing is done after `finish()`, so it would seem to me the right choice would be to make `take_output_stream()` imply `finish()` which would in turn result in error status in case of drawing operations comin in afterwards.